src/samplers/SamplerKronecker/Kronecker.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Coded by Mayur Patel
+ * Copyright (c) 2021
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of this project.
+ */
+
+#include "Kronecker.hpp"
+
+const double Kronecker::R2[2] = {
+  0.75487766624669276,
+  0.56984029099805327
+};
+
+const double Kronecker::R3[3] = {
+  0.81917251339616444,
+  0.67104360670378921,
+  0.54970047790197027
+};
+
+const double Kronecker::R4[4] = {
+  0.85667488385450287,
+  0.73389185662712599,
+  0.62870672103780863,
+  0.53859725722361005
+};
+
+const double Kronecker::K21_2[2] = {
+  std::sqrt((long double)506598872547596.0)/(long double)29147227.0,
```

Comment:
My compiler is complaining about the `long double` to `double` cast in intializer lists. Ok for you if I do explicit cast ?: 

```suggestion
  static_cast<double>(std::sqrt((long double)506598872547596.0)/(long double)29147227.0),
```

Reply:
I updated my branch to do this static cast, thanks.